### PR TITLE
Remove @GameProperty annotation

### LINKED ISSUE #3250: Remove @GameProperty annotation
Now that reflective access to methods annotated with `@GameProperty` has been removed, this annotation is no longer necessary.  However, it does provide some metadata that may be useful, and, if so, that metadata should probably be captured someplace else.

The four pieces of metadata associated with this annotation are described below:

Metadata | Description
:-- | :--
`xmlProperty` | Indicates the property can be set by the `GameParser`, and that it can appear in a map XML file.
`gameProperty` | Indicates the property can be set by an `ObjectPropertyChange` or a `ChangeAttachmentChange` instance.
`adds` | Indicates the property is a collection that will be augmented each time it is set rather than being overwritten with the new values.
`virtual` | Indicates the property does not have a backing field and is only used for convenience.

`virtual` was only added to help `ValidateAttachmentsTest` better isolate which attachment properties needed to be tested.  It no longer has any value since that test has since been removed.

I never quite understood what value `xmlProperty` and `gameProperty` were supposed to provide.

All XML properties should be listed and described in the PoS2 map, so I would think we should be looking at that file as the single source of truth for what constitutes an XML property.

There's currently only six attachment properties that have `gameProperty = false`:

* `AbstractTriggerAttachment#setTrigger()` (DEPRECATED)
* `UnitAttachment#setArtillery()`
* `UnitAttachment#setArtillerySupportable()`
* `UnitAttachment#setUnitSupportCount()`
* `UnitAttachment#setIsParatroop()` (DEPRECATED)
* `UnitAttachment#setIsMechanized()` (DEPRECATED)

For those that aren't deprecated, I'm not even sure that they're _not_ set via a `Change`.  Any insight here?

I think `adds` has some value, as it changes the semantics of how one sets a property (i.e. knowing you may have to reset a property value before setting it).  I think the code makes it clear whether or not a property "adds to" or not.  However, since `adds = true` only appears on the `String` setters (i.e. those called with values read from the XML), this constraint should probably be described in the PoS2 XML.

#### TL;DR:

Without `ValidateAttachmentsTest` to check the semantics of classes that have `@GameProperty` annotations, we risk these annotations (and their metadata) diverging from the implementation of the methods to which they are applied.

Questions:

* Should `@GameProperty` be removed?
* If so, what `@GameProperty` metadata should be retained and where should it be hosted?

### PR DESCRIPTION
**Pull request by ssoloff**
Resolves #3250.

This PR removes all `@GameProperty` annotations and any associated comment that had to do with this annotation (e.g. about the `adds = true` nature of the property) or regarding reflective access to the property.  I also updated a few comments to remove references to the legacy `PropertyUtils` class.

There was a single unused method identified by static analysis after removing the annotation, so I went ahead and removed the method (noted below).  Other than that, there are no code changes in this PR.

### CODE REVIEW

File: game-core/src/main/java/games/strategy/engine/data/Unit.java [hunk 1]
```diff
@@ -58,21 +63,11 @@ public int getHits() {
     return m_hits;
   }
 
-  /**
-   * Remember to always use a ChangeFactory change over an IDelegate Bridge for any changes to game data, or any change
-   * that should go over
-   * the network.
-   */
```
Comment (ssoloff) [inline game-core/src/main/java/games/strategy/engine/data/Unit.java]: This seemed like a useful reminder, so I moved it to the class Javadoc.

File: game-core/src/main/java/games/strategy/engine/data/Unit.java [hunk 2]
```diff
@@ -58,21 +63,11 @@ public int getHits() {
...
   public void setHits(final int hits) {
     m_hits = hits;
   }
 
-  /**
-   * can be null.
-   */
-  @GameProperty(xmlProperty = false, gameProperty = true, adds = false)
-  public void setOwner(final PlayerID player) {
+  public void setOwner(final @Nullable PlayerID player) {
```
Comment (ssoloff) [inline game-core/src/main/java/games/strategy/engine/data/Unit.java]: I replaced the comment with a type annotation.

File: game-core/src/main/java/games/strategy/triplea/attachments/AbstractConditionsAttachment.java [hunk 3]
```diff
@@ -289,13 +279,6 @@ public int getChanceIncrementOnFailure() {
     return m_chanceIncrementOnFailure;
   }
 
-
-  @GameProperty(xmlProperty = true, gameProperty = true, adds = false)
-  private void setChanceDecrementOnSuccess(final String value) {
-    setChanceDecrementOnSuccess(getInt(value));
-  }
```
Comment (ssoloff) [inline game-core/src/main/java/games/strategy/triplea/attachments/AbstractConditionsAttachment.java]: This method was identified by static analysis as unused (presumably the annotation was preventing the static analyzer from identifying that before this change).  I think it became unused when the `MutableProperty#ofMapper()` helper was added.

File: game-core/src/main/java/games/strategy/triplea/attachments/TechAttachment.java [hunk 4]
```diff
@@ -90,12 +89,10 @@ public TechAttachment() {
   }
 
   // setters
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/triplea/attachments/TechAttachment.java]: Side note: This comment should be removed.

**Comment by ssoloff** [comment]
> I hoped we'd map more methods to the ofMapper scheme before this change

@RoiEXLab Sorry, I didn't realize the previous PR was part one of many.  I went back and re-read your comments in #3250 and now realize the connection you were trying to make. :disappointed:   I would have been ok keeping this PR open for a while, as the merge conflicts would be easy to resolve.

However, you should be able to search the diff of this PR for `adds = false` candidates.  That is probably easier than going through each `String` setter individually.

**Pull request merged**